Users/ersciple/m108findandmatch

ericsciple opened the pull request.

TingluoHuang commented on node/task.ts:
```diff
@@ -1094,6 +1095,434 @@ class FindItem {
+            (negateCount % 2 == 1 && matchOptions.flipNegate);
+
+        // set nonegate - brace expansion could result in a leading '!'
+        matchOptions.nonegate = true;
```
set matchOptions.flipNegate=false

TingluoHuang commented on node/task.ts:
```diff
@@ -1094,6 +1095,434 @@ class FindItem {
+        pattern = process.platform == 'win32' ? pattern.replace(/\\/g, '/') : pattern;
+
+        // expand braces - required to accurately interpret findPath
+        let expanded: string[] = matchOptions.nobrace ? [ pattern ] : braceexpansion(pattern);
```
check whether minimatch has minimatch.braceexpansion() function.

TingluoHuang commented on node/task.ts:
```diff
@@ -1094,6 +1095,434 @@ class FindItem {
+                // determine the findPath
+                let findPath = '';
+                let statOnly = false;
+                if (matchOptions.matchBase && !isRooted(pattern) && pattern.indexOf(path.sep) < 0) { // note, relies on assumption '\' have been converted to '/' on Windows
```
path.sep or path.altsep

ericsciple commented:
is findMatch() a better name?

ericsciple commented:
@bryanmacfarlane this is ready now.

i went ahead and removed the glob functions and bumped the version to 2.0.0. i can revert removing the glob() functions if you think that's better at this point.

i also removed some gulp dependencies that dont appear to be used anymore.

i also changed the function name to findMatch (removed "and")

i want to followup with a small markdown discussing the match functions, the two typical usage patterns (i.e. filepath only vs root + text area), recommended default options, and touch lightly on the globbing syntaxes.

i've also started looking into porting to ps3 - will be a separate PR. i'm hopeful it will go fairly quickly.

ericsciple commented:
also, fyi, i reviewed this again today with @TingluoHuang. he and i discussed it extensively at the beginning of the week since then i had integrated good feedback

ericsciple commented:
i found two small issues i need to address before merging (didnt fallback to system.defaultworkingdirectory correctly, and forgot to delete glob functions although removed npm references)
